File: java/duck/ducks/DecoyDuck.java
package ducks;

import behaviors.fly.FlyNoWay;
import behaviors.quack.MuteQuack;

public class DecoyDuck extends Duck {

    public DecoyDuck(String name) {
        super(name);
        flyBehavior = new FlyNoWay(name);
        quackBehavior = new MuteQuack(name);
    }
}

File: kotlin/duck/ducks/DecoyDuck.kt
package ducks

import behaviors.fly.FlyNoWay
import behaviors.quack.MuteQuack

class DecoyDuck(name: String) : Duck(name) {
    init {
        flyBehavior = FlyNoWay(name)
        quackBehavior = MuteQuack(name)
    }
}